### PR TITLE
Closes #179 | Implement `indo_story_cloze` dataloader

### LINKED ISSUE #179: Create dataset loader for indo_story_cloze
Dataloader name: `indo_story_cloze/indo_story_cloze.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?indo_story_cloze

|     Dataset| indo_story_cloze |
|-------------|---|
| Description | This is the first commonsense reasoning dataset in Indonesian. IndoStoryCloze is a story in our dataset that consists of the four-sentence premise, one-sentence correct ending, and one-sentence incorrect ending. In total, we have created 2,325 Indonesian stories with the train/dev/test split 1,000/200/1,135. |
| Subsets | - |
| Languages | ind |
| Tasks | Commonsense Reasoning |
| License | MIT (mit) |
| Homepage | https://huggingface.co/datasets/indolem/indo_story_cloze |
| HF URL | https://huggingface.co/datasets/indolem/indo_story_cloze |
| Paper URL | https://aclanthology.org/2022.csrr-1.2.pdf |

### PR DESCRIPTION
**Pull request by chenxwh**
Closes #179.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/indo_story_cloze/indo_story_cloze.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+}
+
+_SUPPORTED_TASKS = [Tasks.COMMONSENSE_REASONING]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class IndoStoryCloze(datasets.GeneratorBasedBuilder):
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/indo_story_cloze/indo_story_cloze.py:71]:
```suggestion
class IndoStoryClozeDataset(datasets.GeneratorBasedBuilder):
```

File: seacrowd/sea_datasets/indo_story_cloze/indo_story_cloze.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    """IndoStoryCloze is a Story Cloze dataset in Indonesian."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name="indo_story_cloze_source",
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/indo_story_cloze/indo_story_cloze.py:79]: I suggest using `_DATASETNAME` here

**Review by ljvmiranda921 — COMMENTED** [review]
Dataloader works as expected! I just have a few formatting nits.
Also you forgot to add a `__init__.py` file.

**Comment by chenxwh** [comment]
Thanks for the comments! The modifications are incorporated into the latest commit.

**Review by ljvmiranda921 — APPROVED** [review]
LGTM!

**Comment by danjohnvelasco** [comment]
Code is working as expected on jupyter notebook.

However, running the test suite via `python -m tests.test_seacrowd seacrowd/sea_datasets/indo_story_cloze/indo_story_cloze.py` returns the following error: `UnicodeDecodeError: 'charmap' codec can't decode byte 0x9d in position 4415: character maps to <undefined>`. I'm not sure if this only happens on my side.

This can be solved by specifying the encoding when you open the CSV file. For example:
```python
data = csv.DictReader(open(filepath[split], newline="", encoding="utf-8"))
```

Other than this, I have no more issues with the code. Thank you for your work :)

**Comment by chenxwh** [comment]
> Code is working as expected on jupyter notebook.
> 
> However, running the test suite via `python -m tests.test_seacrowd seacrowd/sea_datasets/indo_story_cloze/indo_story_cloze.py` returns the following error: `UnicodeDecodeError: 'charmap' codec can't decode byte 0x9d in position 4415: character maps to <undefined>`. I'm not sure if this only happens on my side.
> 
> This can be solved by specifying the encoding when you open the CSV file. For example:
> 
> ```python
> data = csv.DictReader(open(filepath[split], newline="", encoding="utf-8"))
> ```
> 
> Other than this, I have no more issues with the code. Thank you for your work :)

Thank you, added those!

**Comment by ljvmiranda921** [comment]
LGTM! Wanna review this again, @danjohnvelasco ?

**Comment by chenxwh** [comment]
Can this be closed and merged now?

**Comment by danjohnvelasco** [comment]
LGTM! Merging this now.

**Pull request merged**